libs/external-db-postgres/src/postgres_data_provider.ts

```diff
@@ -1,33 +1,39 @@
-const { escapeIdentifier, prepareStatementVariables  } = require('./postgres_utils')
-const { asParamArrays, patchDateTime, updateFieldsFor } = require('@wix-velo/velo-external-db-commons')
-const { translateErrorCodes } = require('./sql_exception_translator')
+import { Pool } from 'pg'
+import { escapeIdentifier, prepareStatementVariables } from './postgres_utils'
+import { asParamArrays, patchDateTime, updateFieldsFor } from '@wix-velo/velo-external-db-commons'
+import { translateErrorCodes } from './sql_exception_translator'
+import { IDataProvider, AdapterFilter, Sort, Item, AdapterAggregation as Aggregation, AnyFixMe } from '@wix-velo/velo-external-db-types'
+import FilterParser from './sql_filter_transformer'
 
-class DataProvider {
-    constructor(pool, filterParser) {
+export default class DataProvider implements IDataProvider {
+    pool: Pool
+    filterParser: FilterParser
+
+    public constructor(pool: Pool, filterParser: FilterParser) {
         this.filterParser = filterParser
         this.pool = pool
     }
 
-    async find(collectionName, filter, sort, skip, limit, projection) {
+    async find(collectionName: string, filter: AdapterFilter, sort: Sort[], skip: number, limit: number, projection: string[]): Promise<Item[]> {
         const { filterExpr, parameters, offset } = this.filterParser.transform(filter)
         const { sortExpr } = this.filterParser.orderBy(sort)
         const projectionExpr = this.filterParser.selectFieldsFor(projection)
-        const resultset = await this.pool.query(`SELECT ${projectionExpr} FROM ${escapeIdentifier(collectionName)} ${filterExpr} ${sortExpr} OFFSET $${offset} LIMIT $${offset + 1}`, [...parameters, skip, limit])
+        const resultSet = await this.pool.query(`SELECT ${projectionExpr} FROM ${escapeIdentifier(collectionName)} ${filterExpr} ${sortExpr} OFFSET $${offset} LIMIT $${offset + 1}`, [...parameters, skip, limit])
                                     .catch( translateErrorCodes )
-        return resultset.rows
+        return resultSet.rows
     }
 
-    async count(collectionName, filter) {
+    async count(collectionName: string, filter: AdapterFilter) {
```

Comment:
I think its better to import AdapterFilter as Filter to its a little bit confusing